src/Logger.jl

```diff
@@ -111,3 +111,64 @@ function reset_tb_logs()
     logdir = default_logging_session[].logdir
     default_logging_session[] = Logger(logdir, overwrite=true)
 end
+
+
+# Implement the AbstractLogger Interface
+
+catch_exceptions(lg::TBLogger) = false
+
+min_enabled_level(lg::TBLogger) = lg.min_level
+
+# For now, log everything that is above the lg.min_level
+shouldlog(lg::TBLogger, level, _module, group, id) = true
+
+function handle_message(lg::TBLogger, level, message, _module, group, id, file, line; kwargs...)
+    # Unpack the message
+    summ    = SummaryCollection()
+    i_step = 1 # :log_step_increment default value
+
+    if !isempty(kwargs)
+        data = Vector{Pair{String,Any}}()
+
+        # ∀ (k-v) pairs, decompose values into objects that can be serialized
+        for (key,val) in pairs(kwargs)
+            # special key describing step increment
+            if key == :log_step_increment
+                i_step = val
+                continue
+            end
+
+            preprocess(message*"/$key", val, data)
+        end
+
+        # Serialize every object
+        for (name,val) in data
+            push!(summ.value, summary_impl(name, val))
+        end
+    end
+    iter = increment_step(lg, i_step)
+    write_event(lg.file, make_event(lg, summ, step=iter))
+end
+
+"""
+    preprocess(name, val, data)
+
+This method takes a tag `name` and the value `val::T` pair. If type `T` can be
+serialized to TensorBoard then the pair is pushed to `data`, otherwise it should
+call `preprocess` recursively with some simpler types, until a serializable
+type is finally hit.
+
+For a struct, it calls preprocess on every field.
+"""
+function preprocess(name, val::T, data) where T
+    if isstructtype(T)
+        fn = fieldnames(T)
+        for f=fn
+            prop = getproperty(val, f)
+            preprocess(name*"/$f", val, data)
+        end
+    else
+        throw(ErrorException("Can't log type $T, but can't preprocess it either.\n You should define preprocess(name, val::$T, data)."))
```

Comment:
In the future this will fallback to a text logger,

Reply:
for now i think an error is fine, but exprience may end up suggestuing that until we have a text logger fallback we might be better to silently drop it.

Reply:
but I am happy with an error for now

Reply:
I agree. 
Silent dropping is indeed better ATM so that that the package is already somewhat usable.